src/components/exit-soknad/DeleteProsessModal.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useEffect } from "react";
+import Link from "next/link";
+import { Alert, Button, Heading, Modal } from "@navikt/ds-react";
+import { useSanity } from "../../context/sanity-context";
+import { useUuid } from "../../hooks/useUuid";
+import { useDeleteRequest } from "../../hooks/useDeleteRequest";
+import styles from "./ExitSoknad.module.css";
+import { QuizProsess } from "../../types/quiz.types";
+
+interface IProps {
+  isOpen: boolean;
+  handleClose: () => void;
+  prosessType: QuizProsess;
+}
+
+export function DeleteProsessModal({ isOpen, handleClose, prosessType }: IProps) {
+  const { uuid } = useUuid();
+  const { getAppText } = useSanity();
+  const [deleteProsess, deleteProsessStatus, , resetDeleteProsessError] =
+    useDeleteRequest("soknad/delete");
+
+  useEffect(() => {
+    if (Modal.setAppElement) {
+      Modal.setAppElement("#__next");
+    }
+  }, []);
+
+  function closeModal() {
+    if (deleteProsessStatus === "error") {
+      resetDeleteProsessError();
+    }
+    handleClose();
+  }
+
+  return (
+    <>
+      {deleteProsessStatus === "success" && (
```

Comment:
Et notat mer enn noe annet. Denne måten å inkludere modaler på er ikke anbefalt av react-modal, siden `deleteProsessStatus` kan oppdatere seg før lifecycle-endringene til modalen har kjørt seg ferdige. Jeg merket dette da overlayen aldri ble borte når jeg lukket modalen igjen. Noe å være obs på og teste på, spesielt hvis man ikke redirecter brukeren til en annen side når modalen lukkes.

Reply:
Med overlay mener jeg at scroll på siden ble disablet, og aldri enablet igjen. Noe å teste på :)

Reply:
Hmm, vi burde kanskje flytte if-en til inn til Modal.content da så den blir mountet?

Reply:
Jepp! Jeg anbefaler å benytte `open` og `onClose` for å kalkulere dette, siden det er de som ser ut til å styre lifecycle-endringene som må gjøres i modalen. Det er slik jeg har fikset det for dokumentkrav, denne komponenten rendres alltid ut, og `open` styrer om modalen skal vises: https://github.com/navikt/dp-soknadsdialog/blob/main/src/components/dokumentkrav/DokumentkravBundleErrorModal.tsx#L66